### DISCUSSION TITLE
The "Strict-Transport-Security" HTTP header is not set to at least "15552000" seconds.

### DISCUSSION THREAD

**Discussion by vonkraus** [NginxProxyManager/nginx-proxy-manager#3499] [Q&A]
Hello, I apologize if this has been asked before, I searched throughout this git to see if there was anything related, but didn't find anything.

I'm running Unraid 6.12.6 with Nginx for my proxy to my website for Nextcloud.  Checking the Nextcloud security and setup warnings page, I get the warning - The "Strict-Transport-Security" HTTP header is not set to at least "15552000" seconds.  The docs for nextcloud tell me to set this in the nginx.conf file.  I do set these values are restart both NC and Nginx containers on Unraid, but nothing changes and keep getting the same error.  Anyone know where I'm going wrong?

Thanks in advance!

**Comment by blaine07** [comment]
In Nextcloud appdata folder - there is a ssl.conf - do you have it enable/disabled there?
![image](https://github.com/NginxProxyManager/nginx-proxy-manager/assets/63940698/974ed1ac-6de7-43f5-8c74-e265399e68c8)

Line 21:
![image](https://github.com/NginxProxyManager/nginx-proxy-manager/assets/63940698/4350fa8b-c437-41f2-8f5f-c2ea85b022bf)

**Reply by vonkraus** [reply]
I suppose I should have mentioned the rev of nextcloud.  I've tried the following versions in the apps "store" knex666, Linux server, kilrahs and none of them have an ssl.conf file.  They used to as that is where I configured this setting before.  (I've been comparing truenas and unraid and I'm going with unraid) 
What's the recommended nextcloud rev to use?

**Reply by blaine07** [reply]
LinuxServer.IO's is the ONE to use IMO.

**Reply by vonkraus** [reply]
How would I get ffmpeg support in the linuxserv version?

**Reply by vonkraus** [reply]
I think the memories app inside of Nextcloud adds that function...

**Comment by vonkraus** [comment]
I have an update.  If you go into the following path and add the recommendations to your proxy conf files; to correct the red flags from a scan of your website from this scan tool: https:// securityheaders dot com/  This is the way I was able to fix the STS Header errors.
![nginxss](https://github.com/NginxProxyManager/nginx-proxy-manager/assets/57960658/59d194b9-d5f6-4745-a3e6-caa7703fe8aa)

**Comment by ARVEDz** [comment]
I have to open this again, unfortunately. I get an A+ on securityheaders.com but my internal security check in nextcloud still states:

> Some headers are not set correctly on your instance - The `Strict-Transport-Security` HTTP header is not set (should be at least `15552000` seconds). For enhanced security, it is recommended to enable HSTS. For more details see the [documentation ↗](https://docs.nextcloud.com/server/31/go.php?to=admin-security).

I am running nextcloud via docker with Nginx Proxy Manager.

Any Idea or tips, what I should check?

Thanks in advance

Best Regards
Arved

**Reply by drybx** [reply]
Did you figure it out in the meantime? I'm having the same issue

**Reply by popy2k14** [reply]
Same issue here, any hints?

**Reply by TomSaw** [reply]
Me too, sad times :(

**Comment by rmlanghopto** [comment]
If you add:

`Header always set Strict-Transport-Security "max-age=15552000; includeSubDomains"`

under the `<IfModule mod_env.c>` section of the .htaccess in your root Nextcloud folder, that worked for me.

Edit: I'm using Nextcloud's official, stable image in Docker with NPM proxying it.

**Reply by drybx** [reply]
> If you add:
> 
> `Header always set Strict-Transport-Security "max-age=15552000; includeSubDomains"`
> 
> under the `<IfModule mod_env.c>` section of the .htaccess in your root Nextcloud folder, that worked for me.
> 
> Edit: I'm using Nextcloud's official, stable image in Docker with NPM proxying it.

That worked for me. Thanks!
